/*
 * Copyright (c) 2014-2016 - Adjacent Link LLC, Bridgewater, New Jersey
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 *
 *  * Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 *  * Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in
 *    the documentation and/or other materials provided with the
 *    distribution.
 *  * Neither the name of Adjacent Link LLC nor the names of its
 *    contributors may be used to endorse or promote products derived
 *    from this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
 * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
 * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
 * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
 * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
 * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGE.
 *
 * See toplevel COPYING for more information.
 */

 /*
  * !!!
  * !!! THIS FILE IS AUTOMATICALLY GENERATED. DO NOT MODIFY
  * !!!
  */

syntax = "proto2";

package OpenTestPoint;

option optimize_for = SPEED;

import "measurementtable.proto";


message Measurement_emane_ieee80211abg_counters_general
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_counters_general"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

  optional double avgprocessapiqueuedepth = 2 ;
  optional double avgprocessapiqueuewait = 3 ;
  optional double avgtimedeventlatency = 4 ;
  optional double avgtimedeventlatencyratio = 5 ;
  optional uint32 numbroadcastbytesunsupported = 6 ;
  optional uint32 numbroadcastpacketsunsupported = 7 ;
  optional uint32 numdownstreambroadcastdatadiscardduetotxop = 8 ;
  optional uint32 numdownstreamunicastdatadiscardduetoretries = 9 ;
  optional uint32 numdownstreamunicastdatadiscardduetotxop = 10 ;
  optional uint32 numdownstreamunicastrtsctsdatadiscardduetoretries = 11 ;
  optional uint32 numonehopnbrhighwatermark = 12 ;
  optional uint32 numrxonehopnbrlistevents = 13 ;
  optional uint32 numrxonehopnbrlistinvalidevents = 14 ;
  optional uint32 numtwohopnbrhighwatermark = 15 ;
  optional uint32 numtxonehopnbrlistevents = 16 ;
  optional uint32 numunicastbytesunsupported = 17 ;
  optional uint32 numunicastpacketsunsupported = 18 ;
  optional uint32 numupstreambroadcastdatadiscardduetoclobberrxduringtx = 19 ;
  optional uint32 numupstreambroadcastdatadiscardduetoclobberrxhiddenbusy = 20 ;
  optional uint32 numupstreambroadcastdatadiscardduetosinr = 21 ;
  optional uint32 numupstreambroadcastdatanoisehiddenrx = 22 ;
  optional uint32 numupstreambroadcastdatanoiserxcommon = 23 ;
  optional uint32 numupstreamunicastdatadiscardduetoclobberrxduringtx = 24 ;
  optional uint32 numupstreamunicastdatadiscardduetoclobberrxhiddenbusy = 25 ;
  optional uint32 numupstreamunicastdatadiscardduetosinr = 26 ;
  optional uint32 numupstreamunicastdatanoisehiddenrx = 27 ;
  optional uint32 numupstreamunicastdatanoiserxcommon = 28 ;
  optional uint32 numupstreamunicastrtsctsdatarxfromphy = 29 ;
  optional uint32 numupstreamunicastrtsctsrxfromphy = 30 ;
  optional uint64 processedconfiguration = 31 ;
  optional uint64 processeddownstreamcontrol = 32 ;
  optional uint64 processeddownstreampackets = 33 ;
  optional uint64 processedevents = 34 ;
  optional uint64 processedtimedevents = 35 ;
  optional uint64 processedupstreamcontrol = 36 ;
  optional uint64 processedupstreampackets = 37 ;
}

message Measurement_emane_ieee80211abg_counters_queue0
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_counters_queue0"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

  optional float avgdownstreamprocessingdelay0 = 2 ;
  optional float avgupstreamprocessingdelay0 = 3 ;
  optional uint32 numbroadcastbytestoolarge0 = 4 ;
  optional uint32 numbroadcastpacketstoolarge0 = 5 ;
  optional uint64 numdownstreambytesbroadcastgenerated0 = 6 ;
  optional uint64 numdownstreambytesbroadcastrx0 = 7 ;
  optional uint64 numdownstreambytesbroadcasttx0 = 8 ;
  optional uint64 numdownstreambytesunicastgenerated0 = 9 ;
  optional uint64 numdownstreambytesunicastrx0 = 10 ;
  optional uint64 numdownstreambytesunicasttx0 = 11 ;
  optional uint64 numdownstreampacketsbroadcastdrop0 = 12 ;
  optional uint64 numdownstreampacketsbroadcastgenerated0 = 13 ;
  optional uint64 numdownstreampacketsbroadcastrx0 = 14 ;
  optional uint64 numdownstreampacketsbroadcasttx0 = 15 ;
  optional uint64 numdownstreampacketsunicastdrop0 = 16 ;
  optional uint64 numdownstreampacketsunicastgenerated0 = 17 ;
  optional uint64 numdownstreampacketsunicastrx0 = 18 ;
  optional uint64 numdownstreampacketsunicasttx0 = 19 ;
  optional uint32 numhighwatermark0 = 20 ;
  optional uint32 numhighwatermax0 = 21 ;
  optional uint32 numunicastbytestoolarge0 = 22 ;
  optional uint32 numunicastpacketstoolarge0 = 23 ;
  optional uint64 numupstreambytesbroadcastrx0 = 24 ;
  optional uint64 numupstreambytesbroadcasttx0 = 25 ;
  optional uint64 numupstreambytesunicastrx0 = 26 ;
  optional uint64 numupstreambytesunicasttx0 = 27 ;
  optional uint64 numupstreampacketsbroadcastdrop0 = 28 ;
  optional uint64 numupstreampacketsbroadcastrx0 = 29 ;
  optional uint64 numupstreampacketsbroadcasttx0 = 30 ;
  optional uint64 numupstreampacketsunicastdrop0 = 31 ;
  optional uint64 numupstreampacketsunicastrx0 = 32 ;
  optional uint64 numupstreampacketsunicasttx0 = 33 ;
}

message Measurement_emane_ieee80211abg_counters_queue1
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_counters_queue1"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

  optional float avgdownstreamprocessingdelay1 = 2 ;
  optional float avgupstreamprocessingdelay1 = 3 ;
  optional uint32 numbroadcastbytestoolarge1 = 4 ;
  optional uint32 numbroadcastpacketstoolarge1 = 5 ;
  optional uint64 numdownstreambytesbroadcastgenerated1 = 6 ;
  optional uint64 numdownstreambytesbroadcastrx1 = 7 ;
  optional uint64 numdownstreambytesbroadcasttx1 = 8 ;
  optional uint64 numdownstreambytesunicastgenerated1 = 9 ;
  optional uint64 numdownstreambytesunicastrx1 = 10 ;
  optional uint64 numdownstreambytesunicasttx1 = 11 ;
  optional uint64 numdownstreampacketsbroadcastdrop1 = 12 ;
  optional uint64 numdownstreampacketsbroadcastgenerated1 = 13 ;
  optional uint64 numdownstreampacketsbroadcastrx1 = 14 ;
  optional uint64 numdownstreampacketsbroadcasttx1 = 15 ;
  optional uint64 numdownstreampacketsunicastdrop1 = 16 ;
  optional uint64 numdownstreampacketsunicastgenerated1 = 17 ;
  optional uint64 numdownstreampacketsunicastrx1 = 18 ;
  optional uint64 numdownstreampacketsunicasttx1 = 19 ;
  optional uint32 numhighwatermark1 = 20 ;
  optional uint32 numhighwatermax1 = 21 ;
  optional uint32 numunicastbytestoolarge1 = 22 ;
  optional uint32 numunicastpacketstoolarge1 = 23 ;
  optional uint64 numupstreambytesbroadcastrx1 = 24 ;
  optional uint64 numupstreambytesbroadcasttx1 = 25 ;
  optional uint64 numupstreambytesunicastrx1 = 26 ;
  optional uint64 numupstreambytesunicasttx1 = 27 ;
  optional uint64 numupstreampacketsbroadcastdrop1 = 28 ;
  optional uint64 numupstreampacketsbroadcastrx1 = 29 ;
  optional uint64 numupstreampacketsbroadcasttx1 = 30 ;
  optional uint64 numupstreampacketsunicastdrop1 = 31 ;
  optional uint64 numupstreampacketsunicastrx1 = 32 ;
  optional uint64 numupstreampacketsunicasttx1 = 33 ;
}

message Measurement_emane_ieee80211abg_counters_queue2
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_counters_queue2"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

  optional float avgdownstreamprocessingdelay2 = 2 ;
  optional float avgupstreamprocessingdelay2 = 3 ;
  optional uint32 numbroadcastbytestoolarge2 = 4 ;
  optional uint32 numbroadcastpacketstoolarge2 = 5 ;
  optional uint64 numdownstreambytesbroadcastgenerated2 = 6 ;
  optional uint64 numdownstreambytesbroadcastrx2 = 7 ;
  optional uint64 numdownstreambytesbroadcasttx2 = 8 ;
  optional uint64 numdownstreambytesunicastgenerated2 = 9 ;
  optional uint64 numdownstreambytesunicastrx2 = 10 ;
  optional uint64 numdownstreambytesunicasttx2 = 11 ;
  optional uint64 numdownstreampacketsbroadcastdrop2 = 12 ;
  optional uint64 numdownstreampacketsbroadcastgenerated2 = 13 ;
  optional uint64 numdownstreampacketsbroadcastrx2 = 14 ;
  optional uint64 numdownstreampacketsbroadcasttx2 = 15 ;
  optional uint64 numdownstreampacketsunicastdrop2 = 16 ;
  optional uint64 numdownstreampacketsunicastgenerated2 = 17 ;
  optional uint64 numdownstreampacketsunicastrx2 = 18 ;
  optional uint64 numdownstreampacketsunicasttx2 = 19 ;
  optional uint32 numhighwatermark2 = 20 ;
  optional uint32 numhighwatermax2 = 21 ;
  optional uint32 numunicastbytestoolarge2 = 22 ;
  optional uint32 numunicastpacketstoolarge2 = 23 ;
  optional uint64 numupstreambytesbroadcastrx2 = 24 ;
  optional uint64 numupstreambytesbroadcasttx2 = 25 ;
  optional uint64 numupstreambytesunicastrx2 = 26 ;
  optional uint64 numupstreambytesunicasttx2 = 27 ;
  optional uint64 numupstreampacketsbroadcastdrop2 = 28 ;
  optional uint64 numupstreampacketsbroadcastrx2 = 29 ;
  optional uint64 numupstreampacketsbroadcasttx2 = 30 ;
  optional uint64 numupstreampacketsunicastdrop2 = 31 ;
  optional uint64 numupstreampacketsunicastrx2 = 32 ;
  optional uint64 numupstreampacketsunicasttx2 = 33 ;
}

message Measurement_emane_ieee80211abg_counters_queue3
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_counters_queue3"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

  optional float avgdownstreamprocessingdelay3 = 2 ;
  optional float avgupstreamprocessingdelay3 = 3 ;
  optional uint32 numbroadcastbytestoolarge3 = 4 ;
  optional uint32 numbroadcastpacketstoolarge3 = 5 ;
  optional uint64 numdownstreambytesbroadcastgenerated3 = 6 ;
  optional uint64 numdownstreambytesbroadcastrx3 = 7 ;
  optional uint64 numdownstreambytesbroadcasttx3 = 8 ;
  optional uint64 numdownstreambytesunicastgenerated3 = 9 ;
  optional uint64 numdownstreambytesunicastrx3 = 10 ;
  optional uint64 numdownstreambytesunicasttx3 = 11 ;
  optional uint64 numdownstreampacketsbroadcastdrop3 = 12 ;
  optional uint64 numdownstreampacketsbroadcastgenerated3 = 13 ;
  optional uint64 numdownstreampacketsbroadcastrx3 = 14 ;
  optional uint64 numdownstreampacketsbroadcasttx3 = 15 ;
  optional uint64 numdownstreampacketsunicastdrop3 = 16 ;
  optional uint64 numdownstreampacketsunicastgenerated3 = 17 ;
  optional uint64 numdownstreampacketsunicastrx3 = 18 ;
  optional uint64 numdownstreampacketsunicasttx3 = 19 ;
  optional uint32 numhighwatermark3 = 20 ;
  optional uint32 numhighwatermax3 = 21 ;
  optional uint32 numunicastbytestoolarge3 = 22 ;
  optional uint32 numunicastpacketstoolarge3 = 23 ;
  optional uint64 numupstreambytesbroadcastrx3 = 24 ;
  optional uint64 numupstreambytesbroadcasttx3 = 25 ;
  optional uint64 numupstreambytesunicastrx3 = 26 ;
  optional uint64 numupstreambytesunicasttx3 = 27 ;
  optional uint64 numupstreampacketsbroadcastdrop3 = 28 ;
  optional uint64 numupstreampacketsbroadcastrx3 = 29 ;
  optional uint64 numupstreampacketsbroadcasttx3 = 30 ;
  optional uint64 numupstreampacketsunicastdrop3 = 31 ;
  optional uint64 numupstreampacketsunicastrx3 = 32 ;
  optional uint64 numupstreampacketsunicasttx3 = 33 ;
}

message Measurement_emane_ieee80211abg_tables_events
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_tables_events"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable eventreceptiontable = 2 ;
}

message Measurement_emane_ieee80211abg_tables_neighbor
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_tables_neighbor"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable neighbormetrictable = 2 ;
   optional MeasurementTable neighborstatustable = 3 ;
   optional MeasurementTable onehopneighbortable = 4 ;
   optional MeasurementTable twohopneighbortable = 5 ;
}

message Measurement_emane_ieee80211abg_tables_status_queue0
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_tables_status_queue0"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable broadcastpacketaccepttable0 = 2 ;
   optional MeasurementTable broadcastpacketdroptable0 = 3 ;
   optional MeasurementTable unicastpacketaccepttable0 = 4 ;
   optional MeasurementTable unicastpacketdroptable0 = 5 ;
}

message Measurement_emane_ieee80211abg_tables_status_queue1
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_tables_status_queue1"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable broadcastpacketaccepttable1 = 2 ;
   optional MeasurementTable broadcastpacketdroptable1 = 3 ;
   optional MeasurementTable unicastpacketaccepttable1 = 4 ;
   optional MeasurementTable unicastpacketdroptable1 = 5 ;
}

message Measurement_emane_ieee80211abg_tables_status_queue2
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_tables_status_queue2"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable broadcastpacketaccepttable2 = 2 ;
   optional MeasurementTable broadcastpacketdroptable2 = 3 ;
   optional MeasurementTable unicastpacketaccepttable2 = 4 ;
   optional MeasurementTable unicastpacketdroptable2 = 5 ;
}

message Measurement_emane_ieee80211abg_tables_status_queue3
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_ieee80211abg_tables_status_queue3"];
    required string module = 2 [default = "otestpoint.emane.ieee80211abg"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable broadcastpacketaccepttable3 = 2 ;
   optional MeasurementTable broadcastpacketdroptable3 = 3 ;
   optional MeasurementTable unicastpacketaccepttable3 = 4 ;
   optional MeasurementTable unicastpacketdroptable3 = 5 ;
}
